### PR TITLE
Helm template

### PR DESCRIPTION
**Pull request by weikinhuang**
Signed-off-by: Wei Kin Huang <weikin.huang04@gmail.com>

Helm options:

```
functionNamespace=defaults to the deployed namespace, kube namespace to create function deployments in
async=true/false defaults to false, deploy nats if true
armhf=true/false, defaults to false, use arm images if true (missing images for async)
exposeServices=true/false, defaults to true, will always create `ClusterIP` services, and expose `NodePorts` if true
rbac=true/false defaults to true, if true create roles
```

### CODE REVIEW

**Comment by weikinhuang** [comment]
```bash
# install tiller
helm init --skip-refresh --upgrade

# for default
helm upgrade --install --debug --reset-values --set async=true openfaas helm/

# for non default ns/separate fn namespace
kubectl create ns openfaas
kubectl create ns openfaas-fn

helm upgrade --install --debug --namespace openfaas --reset-values --set async=true --set functionNamespace=openfaas-fn openfaas helm/
```

**Comment by weikinhuang** [comment]
Refs #10

**Comment by alexellis** [comment]
Very cool @weikinhuang !

**Comment by weikinhuang** [comment]
@alexellis if the preferred method is to use helm, then I would suggest removing the vanilla yaml files and update the documentation. Otherwise you can keep both.

File: helm/values.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+armhf: false
+exposeServices: true
+rbac: true
+
+# images
+images:
+  controller: weikinhuang/faas-netes:function-ns
+  controllerArmhf: functions/faas-netesd-armhf
+
+  gateway: functions/gateway:0.6.5
```
Comment (alexellis) [inline helm/values.yaml]: We have 0.6.5 now too for AMD64

File: helm/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: faas-controller
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - services
+    verbs:
+      - get
```
Comment (gardlt) [inline helm/templates/rbac.yaml]: we should try hosting this in the values file to keep the chart as configurable as possible.

Comment (weikinhuang) [reply helm/templates/rbac.yaml]: you can always bind more roles to the `faas-controller` user, this is the minimum permissions to get faas-netes working

**Comment by alexellis** [comment]
I will merge the changes so we can start integrating them. We can update after some more testing / feedback.

@weikinhuang please add a section to the README.md with the notes in your PR here.

File: helm/templates/rbac.yaml (hunk 2 above)
Comment (alexellis) [reply helm/templates/rbac.yaml]: @gardlt Can you help with testing?

**Comment by alexellis** [comment]
@weikinhuang error during testing:

```
root@k8s-17-master:~/faas-netes# helm upgrade --install --debug --reset-values --set async=true openfaas helm/
[debug] Created tunnel using local port: '41016'

[debug] SERVER: "localhost:41016"

Error: UPGRADE FAILED: User "system:serviceaccount:kube-system:default" cannot list configmaps in the namespace "kube-system". (get configmaps)
```

**Comment by weikinhuang** [comment]
@alexellis looks like your user does not have the right permissions

**Pull request merged**